packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -226,18 +232,42 @@ void updateLocalProperties({String projectPath, BuildInfo buildInfo}) {
     settings.writeContents(localProperties);
 }
 
-Future<Null> findAndReplaceVersionProperties({@required BuildInfo buildInfo}) {
-  assert(buildInfo != null, 'buildInfo can\'t be null');
+Future<Null> findAndReplaceVersion() async {
+  return findAndReplaceVersionInBuildGradle(projectPath: '');
+}
+
+@visibleForTesting
+Future<Null> findAndReplaceVersionInBuildGradle({String projectPath}) async {
+  final Status status = logger.startProgress('Set version in app/build.gradle...', expectSlowOperation: true);
   final Completer<Null> completer = new Completer<Null>();
 
+  FlutterManifest flutterManifest;
+  try {
+    flutterManifest = await FlutterManifest.createFromPath(fs.path.join(projectPath, bundle.defaultManifestPath));
+  } catch (error) {
+    printStatus('Error detected in pubspec.yaml:', emphasis: true);
+    printError('$error');
+    completer.complete();
```

Comment:
Why not rethrow?  If their manifest is invalid and can't be parsed, it seems like we should surface that and halt the build.

Reply:
@tvolkert  In my first version, I called `completer.completeError('test error')` with the result that Flutter crashed:
```
$ flutter build apk --build-name=1.0.3 --build-number=3
Update version in pubspec.yaml...                            0,1s
Set version in app/build.gradle...                           0,0s

Oops; flutter has exited unexpectedly: "test error".
Sending crash report to Google.
Crash report sent (report ID: da9511e51bac8867)
Crash report written to /Volumes/Archiv/hello/flutter_01.log;
please let us know at https://github.com/flutter/flutter/issues.
```

How can I break the build without crashing Flutter?

Reply:
You can complete with a [`ToolExit`](https://github.com/flutter/flutter/blob/e06f69e1cdcd1bf8085e9cf7600ffff64f983b79/packages/flutter_tools/lib/src/base/common.dart#L35) error.

Reply:
Note that we _want_ to cause the build to fail, but yes, without the "flutter has exited unexpectedly" flow 😃